include/openssl/kdf.h

```diff
@@ -90,6 +93,19 @@ extern "C" {
             EVP_PKEY_CTX_ctrl_uint64(pctx, -1, EVP_PKEY_OP_DERIVE, \
                             EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES, maxmem_bytes)
 
+# define EVP_PKEY_CTX_set_pbkdf2_md(pctx, md) \
+            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
+                              EVP_PKEY_CTRL_PBKDF2_MD, 0, (void *)(md))
+
+# define EVP_PKEY_CTX_set1_pbkdf2_salt(pctx, salt, saltlen) \
```

Comment:
Hi @levitte, currently at this. Here's something I didn't anticipate: While consolidating all of these, I ran into this ugly issue:

<pre>
include/openssl/kdf.h:19:0: error: "EVP_PKEY_CTRL_MD" redefined [-Werror]
 # define EVP_PKEY_CTRL_MD                       (EVP_PKEY_ALG_CTRL)
 
In file included from crypto/err/err_all.c:21:0:
include/openssl/evp.h:1233:0: note: this is the location of the previous definition
 # define EVP_PKEY_CTRL_MD                1
</pre>

From my perspective, it does make sense to re-use the already-defined CTRL_MD in evp.h -- or is there any difference between EVP_PKEY_CTRL_MD and the corresponding one that's offset by EVP_PKEY_ALG_CTRL? In other words, do we need to distinguish the two or is it okay to fold them into the same value? Because if we need to differentiate, this makes things much more confusing (there'd be EVP_PKEY_CTRL_MD and EVP_PKEY_ALG_CTRL_MD flying around then). So I'd lean towards folding them (but this would mean that the _MD setter would be the odd-one-out in the KDF bunch).